src/praisonai/praisonai/scheduler/async_agent_scheduler.py

```
@@ -0,0 +1,429 @@
+                    # self._total_cost += estimated_cost
+                
+                safe_call(self.on_success, result)
+                # TODO: Add daemon state update from sync version:
+                # self._update_state_if_daemon()
+                return
+                
+            except asyncio.TimeoutError as e:
+                last_exc = e
+                logger.error(f"Async execution timeout on attempt {attempt + 1}: {e}")
+                # Continue to retry logic
+            except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)